ballerina/tests/47_unary_timestamp_client.bal

```diff
@@ -18,37 +18,102 @@ import ballerina/test;
+isolated function testGetGreeting() returns error? {
+    TimestampServiceClient utsClient = check new ("http://localhost:9147");
     Greeting greeting = check utsClient->getGreeting("Hello");
     time:Utc customTime = [1196676930, 0.12];
     Greeting expectedGreeting = {"name": "Hello","time": customTime};
     test:assertEquals(greeting, expectedGreeting);
 }
 
 @test:Config {enable: true}
-isolated function testUnaryTimestampWithBidiGreeting() returns error? {
-    UnaryTimestampServiceClient utsClient = check new ("http://localhost:9147");
+isolated function testExchangeGreeting() returns error? {
+    TimestampServiceClient utsClient = check new ("http://localhost:9147");
     time:Utc customTime = [1228302930, 0.12];
     Greeting expectedGreeting = {"name": "Hello", "time": customTime};
-    Greeting greeting = check utsClient->getBiGreeting(expectedGreeting);
+    Greeting greeting = check utsClient->exchangeGreeting(expectedGreeting);
     test:assertEquals(greeting, expectedGreeting);
 }
 
 @test:Config {enable: true}
-isolated function testUnaryTimestampWithBidiTime() returns error? {
-    UnaryTimestampServiceClient utsClient = check new ("http://localhost:9147");
+isolated function testExchangeTime() returns error? {
+    TimestampServiceClient utsClient = check new ("http://localhost:9147");
     time:Utc sendingTime = check time:utcFromString("2008-12-03T11:15:30.120Z");
-    time:Utc receivedTime = check utsClient->getBiTime(sendingTime);
+    time:Utc receivedTime = check utsClient->exchangeTime(sendingTime);
     time:Utc expectedTime = check time:utcFromString("2012-12-03T11:13:30.472Z");
     test:assertEquals(receivedTime, expectedTime);
 }
 
 @test:Config {enable: true}
-isolated function testUnaryTimestampWithBidiTimeContext() returns error? {
-    UnaryTimestampServiceClient utsClient = check new ("http://localhost:9147");
+isolated function testExchangeTimeContext() returns error? {
+    TimestampServiceClient utsClient = check new ("http://localhost:9147");
     time:Utc sendingTime = check time:utcFromString("2008-12-03T11:15:30.120Z");
-    ContextTimestamp result = check utsClient->getBiTimeContext(sendingTime);
+    ContextTimestamp result = check utsClient->exchangeTimeContext({
+        headers: {},
+        content: sendingTime
+    });
     time:Utc expectedTime = check time:utcFromString("2012-12-03T11:13:30.472Z");
     test:assertEquals(result.content, expectedTime);
 }
+
+@test:Config {enable: true}
+function testTimestampServerStream() returns error? {
+    TimestampServiceClient utsClient = check new ("http://localhost:9147");
+    time:Utc sendingTime = check time:utcFromString("2008-12-03T11:15:30.120Z");
+
+    var result = check utsClient->serverStreamTime(sendingTime);
+    int count = 0;
+    time:Utc[] receivedData = [];
+    error? e = result.forEach(function(time:Utc value) {
+        receivedData[count] = <time:Utc>value;
```

Comment:
Do we need to cast the `value` here?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/372/commits/69f160eca3668e350a01ca297ba3404aae9ce7d2